### PR TITLE
mini pepperball attachment

### PR DESCRIPTION
**Pull request by Pill-U**
## About The Pull Request ##
Adds the mini pepperball gun, an attachable variant of the pepperball gun and a mag for it.
It has a slower rate of fire than the regular version, a smaller magazine size and drains less plasma.
Todo:
~~Can currently shoot while unattached and while single-handed. (you don't want to use it that way anyways since it's just a worse pepperball gun)~~
~~Fire mode can currently only be changed while the gun is unattached. (Possibly remove its autofire)~~
Find more bugs.
~~Remove unnecessary code.~~
~~Sprites. (I'm not a spriter, any help and tips are appreciated)~~ look trash but will do
~~Attachment can currently take regular mags. (But not reverse)~~
Balance the thing.
~~Add it to guns.~~
~~Add it to vendors.~~
Ban it from HvH?

## Why It's Good For The Game
The regular pepperball gun sees next to no use in-game and is more frequently used to commit friendly fire. This aims to make it more useful by letting you use it without the downside of having no gun to do damage with, in exchange for not having a better underbarrel attachment and a smaller magazine capacity and fire rate.
~~BALANCE CONCERN: This gun can currently in theory remove about 2/3rd of any xenos plasma supply if you hit all 20 shots and can fire at the same time as the gun it's attached to.~~ Plasma-B-Gone nerfed but not quite trash.

## Changelog
:cl:
add: Add pepperball attachment
add: Add ammo for the pepperball attachment
add: Sprites for both
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -1564,6 +1564,11 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	icon_state = "flame_wide_red"
 	range_modifier = 0
 
+/obj/item/weapon/gun/rifle/pepperball/pepperball_mini
```
Comment (hyper2snyper) [inline code/modules/projectiles/gun_attachables.dm:1567]: ok so let me explain this. You are defining the gun a second time. in gun_attachables. You dont want to do this. Put the attach delay underneathe the gun and remove this.

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -548,6 +548,20 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 	placed_overlay_iconstate = "pepper"
 
 //-------------------------------------------------------
+
+/obj/item/weapon/gun/rifle/pepperball/pepperball_mini	//TODO: firemode currently has to be changed before attaching the gun
+	name = "mini pepperball gun"						//TODO: gun can currently still be fired unattached
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/specialist.dm:553]: GUN_ATTACHMENT_FIRE_ONLY under flags_gun_features to make sure it only fires when attached.

File: code/modules/projectiles/guns/specialist.dm [hunk 3]
```diff
@@ -548,6 +548,20 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 	placed_overlay_iconstate = "pepper"
 
 //-------------------------------------------------------
+
+/obj/item/weapon/gun/rifle/pepperball/pepperball_mini	//TODO: firemode currently has to be changed before attaching the gun
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/specialist.dm:552]: firemode changing is something i have to add support for. If you are capable thatd be much appreciated.

**Review by hyper2snyper — APPROVED** [review]
This is something ive wanted for a bit. Im super stoked.

**Comment by Kuro020** [comment]
but how does it look

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 4]
```diff
@@ -141,6 +141,8 @@
 			/obj/item/ammo_magazine/flamer_tank/mini = -1,
 			/obj/item/weapon/gun/grenade_launcher/underslung = -1,
 			/obj/item/attachable/motiondetector = -1,
+			/obj/item/weapon/gun/rifle/pepperball/pepperball_mini,
+			/obj/item/ammo_magazine/rifle/pepperball/pepperball_mini,
```
Comment (hyper2snyper) [inline code/game/objects/machinery/vending/marine_vending.dm:145]: you want this to have a value.
```suggestion
			/obj/item/ammo_magazine/rifle/pepperball/pepperball_mini = -1,
```

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 5]
```diff
@@ -141,6 +141,8 @@
 			/obj/item/ammo_magazine/flamer_tank/mini = -1,
 			/obj/item/weapon/gun/grenade_launcher/underslung = -1,
 			/obj/item/attachable/motiondetector = -1,
+			/obj/item/weapon/gun/rifle/pepperball/pepperball_mini,
```
Comment (hyper2snyper) [inline code/game/objects/machinery/vending/marine_vending.dm:144]:
```suggestion
			/obj/item/weapon/gun/rifle/pepperball/pepperball_mini = -1,
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 6]
```diff
@@ -290,5 +290,14 @@
 	default_ammo = /datum/ammo/bullet/pepperball
 	max_rounds = 70
 	w_class = WEIGHT_CLASS_NORMAL
-	gun_type = /obj/item/weapon/gun/rifle/pepperball
+	gun_type = /obj/item/weapon/gun/rifle/pepperball/
```
Comment (hyper2snyper) [inline code/modules/projectiles/magazines/specialist.dm:293]:
```suggestion
	gun_type = /obj/item/weapon/gun/rifle/pepperball
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 7]
```diff
@@ -290,5 +290,14 @@
...
 	icon_state_mini = "mag_pepperball"
+
+/obj/item/ammo_magazine/rifle/pepperball/pepperball_mini
+	name = "small pepperball canister (SAN balls)"			//TODO: pepperball gun can't take mini ammo, but mini can take regular pepperball ammo
+	desc = "A small canister for use with the miniature pepperball gun."
+	icon_state = "pepperball" //TODO: Custom Sprite
+	default_ammo = /datum/ammo/bullet/pepperball/pepperball_mini
+	max_rounds = 20
+	gun_type = /obj/item/weapon/gun/rifle/pepperball/pepperball_mini
+	icon_state_mini = "mag_pepperball" //TODO: Custom Sprite
```
Comment (hyper2snyper) [inline code/modules/projectiles/magazines/specialist.dm:303]: icon_state_mini is just the icon_state for when its inside an ammo box. you probably dont need a custom sprite for this.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

**Comment by Pill-U** [comment]
I already did all of those I just didn't want to push the commit yet because I wanted to commit the sprites for the attachment at the same time. They're just modified versions of the original pepperball gun and Pariah hasn't answered if I can actually use them in 3 days. If he doesn't answer by the time your gun modularization PR gets merged I'll make some new ones or hope to get some made.

**Comment by Pill-U** [comment]
gonna do the sprites tomorrow mb then this damned thing will be done after a 10 days hiatus

**Comment by Pill-U** [comment]
pepperball empty
pepperball loaded
empty and full sprites. Nothing amazing I know.

File: code/modules/projectiles/ammo_datums.dm [hunk 8]
```diff
@@ -2154,11 +2154,16 @@ datum/ammo/bullet/revolver/tp44
 	damage = 70
 	penetration = 0
 	shrapnel_chance = 0
+	var/drain_multiplier = 0.05
```
Comment (hyper2snyper) [inline code/modules/projectiles/ammo_datums.dm:2157]: autodoc

File: code/modules/projectiles/guns/specialist.dm [hunk 9]
```diff
@@ -454,6 +454,25 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
...
+	slot = ATTACHMENT_SLOT_UNDER
+	max_shells = 20
+	default_ammo_type = /obj/item/ammo_magazine/rifle/pepperball/pepperball_mini
+	allowed_ammo_types = list(/obj/item/ammo_magazine/rifle/pepperball/pepperball_mini)
+	force = 5
+	attachable_allowed = list()
+	gun_firemode_list = list(GUN_FIREMODE_AUTOMATIC)
+	flags_gun_features = GUN_IS_ATTACHMENT | GUN_WIELDED_FIRING_ONLY | GUN_ATTACHMENT_FIRE_ONLY | GUN_AMMO_COUNTER
+	fire_delay = 0.2 SECONDS
+	attach_delay = 3
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/specialist.dm:471]:
```suggestion
	attach_delay = 3 SECONDS
```
otherwise its actually 0.3 seconds

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
Looks good. If you could include screenshots of it attached to a gun itd be good.

**Review by Kuro020 — APPROVED** [review]

**Comment by Pill-U** [comment]
image

**Review by BraveMole — APPROVED** [review]

**Pull request merged**